Client/Application/Migration.swift

```diff
@@ -27,7 +27,17 @@ class Migration {
     
     static func moveDatabaseToApplicationDirectory() {
         //Moves Coredata sqlite file from Documents dir to application support dir.
-        DataController.shared.migrateToNewPathIfNeeded()
+        if !Preferences.Migration.dbLocationMigrationCompleted.value {
+            do {
+                Preferences.Migration.dbLocationMigrationRetiresCount.value += 1
+                try DataController.shared.migrateToNewPathIfNeeded()
+            } catch {
+                if Preferences.Migration.dbLocationMigrationRetiresCount.value < 3 {
+                    moveDatabaseToApplicationDirectory()
+                }
+            }
+            Preferences.Migration.dbLocationMigrationCompleted.value = true
```

Comment:
Not sure we want to silently fail after 3 retries, again, will discuss with product on how they'd prefer to handle this.

Reply:
yeah need advice on this. DO we need to show an alert